pmd-core/etc/xslt/pmd-report-v2.xslt

```diff
@@ -0,0 +1,251 @@
+<?xml version="1.0"?>
+<xsl:stylesheet version="1.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+xmlns:scan="http://pmd.sourceforge.net/report/2.0.0"
+  >
+
+<xsl:template match="scan:pmd">
+  <xsl:variable name="version" select="@version" />
+  <xsl:variable name="timestamp" select="@timestamp" />
+  <xsl:variable name="count">0</xsl:variable>
+  <xsl:variable name="total">0</xsl:variable>
+
+
+  <html>
+   <head>
+   <title>PMD Report </title>
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0/dist/css/bootstrap.min.css" integrity="sha384-gH2yIJqKdNHPEq0n4Mqa/HGKIhSkIHeL5AyhkYV8i59U5AR6csBvApHHNl/vI1Bx" crossorigin="anonymous"/>
+<script src="https://code.jquery.com/jquery-3.6.1.min.js" integrity="sha256-o88AwQnZB+VDvE9tvIXrMQaPlFFSUTR+nldQm1LuPXQ=" crossorigin="anonymous"></script>
+ 
+<link rel="stylesheet" type="text/css" href="https://cdn.datatables.net/v/dt/jszip-2.5.0/dt-1.12.1/b-2.2.3/b-colvis-2.2.3/b-html5-2.2.3/b-print-2.2.3/datatables.min.css"/>
+ 
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/pdfmake.min.js"></script>
+<script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/pdfmake/0.1.36/vfs_fonts.js"></script>
+<script type="text/javascript" src="https://cdn.datatables.net/v/dt/jszip-2.5.0/dt-1.12.1/b-2.2.3/b-colvis-2.2.3/b-html5-2.2.3/b-print-2.2.3/datatables.min.js"></script>
+
+
+    <script src="https://cdn.jsdelivr.net/npm/vega@5.21.0"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-lite@5.2.0"></script>
+    <script src="https://cdn.jsdelivr.net/npm/vega-embed@6.20.2"></script>
+
+    <script>
+      let total = 0;
+      let p1 = 0 , p2 = 0, p3 = 0, p4 = 0, p5 = 0;
+      let bp =0, codeStyle = 0, design = 0, doc = 0, ep = 0, perf = 0, security = 0;
+    </script>
+   
+   </head>
+  <body>
+ <nav class="nav navbar navbar-expand-md navbar-dark bg-dark fixed-top">
+
+ 
+  <div class="container-fluid">
+    <a class="navbar-brand" href="#">PMD Report</a>
+    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+      <span class="navbar-toggler-icon"></span>
+    </button>
+  
+  </div>
+</nav>
+  <div class='container' style='margin-top:100px;'>
+    <h5>Problems found ::  <small> Version: <xsl:value-of select="$version"/> @ <xsl:value-of select="$timestamp"/></small>
+     <span id="summary"></span>  </h5>
+    <table id='pmdTable' class='table table-hover table-striped'>
+    <thead>
+ 
+      <tr>
+        <th>File #</th>
+        <th>Issue #</th>
+        <th>File</th>
+        <th>Line</th>
+        <th>Priority</th>
+        <th>Rule</th>
+        <th>Problem</th>
+
+   
+      </tr>
+      </thead>
+      <xsl:for-each select="scan:file">
+      <xsl:variable name="filename" select="@name" />
+      <xsl:variable name="slno" select="position()" />
```

Comment:
Please give this variable a better name - I don't know what "slno" is. Maybe "fileNumber"?